src/TransformOperationExecutor.ts

```diff
@@ -598,7 +598,35 @@ export class TransformOperationExecutor {
       );
       if (excludedProperties.length > 0) {
         keys = keys.filter(key => {
-          return !excludedProperties.includes(key);
+          const excludeMetadata = defaultMetadataStorage.findExcludeMetadata(
+            target,
+            key
+          );
+
+          let shouldExclude = excludedProperties.includes(key);
+
+          if(!excludeMetadata || !excludeMetadata.options) {
+            return !shouldExclude;
+          }
+         
+          // apply grouping exclusion options
+          if(excludeMetadata.options.groups) {
+            shouldExclude = this.options.groups && this.options.groups.length
+              ? this.checkGroups(excludeMetadata.options.groups)
+              : false;
+          }
+
+          // apply versioning exclusion options
+          if(excludeMetadata.options.since || excludeMetadata.options.until) {
+            shouldExclude = this.options.version
+              ? this.checkVersion(
+                  excludeMetadata.options.since, 
+                  excludeMetadata.options.until
+                )
+              : false;            
+          }
+         
+          return !shouldExclude;
```

Comment:
If I read this correctly, it is going to include the property if group check fails, but version check passes.

```ts
const version = 2

class Item {
  @Exclude({ group: ['group'], since: 3 })
  id: number;
}

const item = new Item()
item.id = 1

console.log(classToPlain(item)) // Item { id: 1 }
```

Reply:
I understand this is a design choice. 

This is still much better than not having the ability to exclude properties by group. I am absolutely positive it should be merged even if this behaviour does not satisfy all needs, make class-transformer better one step at the time. That said, let me introduce a real life example of an issue we face in current project:

We want to share a limited representation of our object with a 3rd party integration:

```ts
class Ticket {
  @Exclude({ group: ['external'] })
  priority: number;
}
```

Now in the future version of the API, we may want to deprecate the `priority` altogether:

```ts
class Ticket {
  @Exclude({ group: ['external'], since: 2 })
  priority: number;
}
```

With current logic in place, due to AND conjunction, this would expose `priority` for API v1 group `external`.

The problem can be solved in multitude of different ways, this is likely not the place to discuss them, I just wanted to clarify current implementation is a design choice and highlight it's downfalls.

Reply:
Actually, it's not an AND conjuction. It's more that `version` takes precedence over `groups`. 

```typescript
class Ticket {
  @Exclude({ groups: ['external'], since: 2 })
  priority: number;
}
```
In v1, it would be exposed for everyone
```typescript
classToPlain(ticker, { groups: ['external'], version: 1 });
// {priority: 1}
classToPlain(ticker, { version: 1 });
// {priority: 1}
```

Since v2 it will be excluded for all groups, because version takes precedence over groups:
```typescript
classToPlain(ticker, { groups: ['external'], version: 2 });
// {}
classToPlain(ticker, { version: 2 });
// {}
```

Maybe it should be *at least* an AND conjuction. I'm not sure how `Expose` works in this situation, I haven't used groups and version together. I will take a look at `Expose` and do some testing tomorrow (it's almost 12pm here 😂)